### PR TITLE
Explicit caveat about gaps in the docs

### PR DESCRIPTION
**Pull request by jcrben**
I've looked hard for the documentation on a few operators and I can't seem to find it: pretty sure it isn't there. I wouldn't mind (sort of; I'm new to this system and quite busy) adding actual documentation but I'm not sure how many gaps there are and what the project's preference is on documentation minimalism.

A few weeks ago someone [asked about the [`^` operator](https://github.com/wycats/handlebars-site/issues/107), a week ago someone asked about the [`&` operator](https://github.com/wycats/handlebars.js/issues/1041), and [similar questions](http://stackoverflow.com/questions/27026519/handlebars-templates-using-ampersand-and-caret) came up on StackOverflow a couple times in the past year.

It's better to be explicit about gaps in the documentation. Hope you don't take this as being just negative or hypercritical.

### CODE REVIEW

**Comment by kpdecker** [comment]
Glad to accept PRs that help document the gaps as you see them. This is not a productive change IMO.

**Comment by jcrben** [comment]
Is it really that bad to leave this open for discussion for a little while longer? Any reasons why we can't alert readers that this is already a known issue so they know where to look? 

I'm not sure how many collaborators there are but it would be nice to hear other people's thoughts.

**Comment by kpdecker** [comment]
The most helpful thing would be to to file PRs that document the things you see as missing. Barring that, the second most helpful thing to do would be to file an issue highlighting the deficiencies as you see them.

Regarding collaborators, I'm pretty much the only one who does any work on tasks like documentation, etc and I suck at that since I am very familiar with the code and don't know what is clear or not and I also have no incentive to read the docs in my normal usage. Any constructive help is greatly appreciated.

**Comment by jcrben** [comment]
Is it worthwhile to start an issue which is used to list any and all undocumented expressions and API calls? Then we can check them off as these are documented. I noticed that the issue #107 was closed and I'm not sure what came from the investigation, but that's one of the undocumented ones.

I get the sense from certain projects that they basically prefer minimal docs. There is such a thing as being too wordy, which is why my preference is to see a split between a concise narrative introduction and a comprehensive API reference. With handlebars, the API reference doesn't include any expressions, and things are mostly documented narratively. Handlebars and jade seem to be the most popular templating libraries in the JS world which is why I'm particularly interested in improving the docs. Plus I use it for my Ghost blog.

**Comment by kpdecker** [comment]
Yes, we certainly should have a "this feature needs to be documented" sort of listing or individual issues reported even. I'll openly admit I suck at documenting things, particularly features that were around prior to me taking over the project, so I need all the help that I can get.
